### ISSUE TITLE
Create new content type and views for models

### ISSUE THREAD

**Issue by drozas** [CRESS-Surrey/cress_website_code#46]
hi David

The new CRESS site looks great. I will get all the TELL ME materials uploaded and linked in the next week or so. In the meantime, someone asked me about Nigel's housing model. It was on the old site but I can't find it on the new site. This reminded me though, I think it would be good to have a 'models' section of the site. I know we can now upload models as part of the project description, but I think it would be good to have a single place to publish all our models, like the publications page.

Jen

**Comment by drozas** [comment]
Hello Jen, Nigel,

>>> The new CRESS site looks great.

Thanks, Jen. I am really glad you like it :-)

>>> I think it would be good to have a 'models' section of the site. I know we can now upload models as part of the project description, but I think it would be good to have a single place to publish all our models, like the publications page.

Yes, I think that is a good idea. I will need both of you to refine a bit the requirements, however. In architectural terms, I think the best way is to gather it as a content type (as for instance the events, the books, etc.) editable by all CRESS members. We need to clarify a bit more which fields will be needing. Here is a proposal:

    Title: textbox, also employed automatically for the automatically generated URLs.
    Description: textbox, with a WYSIWYG (similar to http://cress.soc.surrey.ac.uk/web/node/add/link)
    Screenshot: maybe including an optional screenshot of the model?
    File: a field to upload the file. What are the expected file types? Do you usually provide it in a single .zip/.tar.gz file? What is the typical size? Should we make this field optional? (in case we have the option to provide a link to an external repository).
    External repository: a field for a link in an external repository (e.g.: GitHub).

Regarding the visualisation, I could create a view (query to the DB) presenting them, and then provide a link for further information. A way to present this could be a list of teasers (as in http://cress.soc.surrey.ac.uk/web/projects), a table (as in http://cress.soc.surrey.ac.uk/web/events), etc. Please, let me know which format do you think is more appropriate. Another issue is: how are we planning to sort them? Alphabetically? By date of creation?


Finally, we need to decide in which section of the menu we will be presenting this. The "gallery" section seems to have too little content from my point of view for being a primary one. Therefore, an alternative could be to create a new landing page: "Downloads" or "Resources" (in a similar way as we did for Teaching - http://cress.soc.surrey.ac.uk/web/teaching), and then having Models and Gallery as children of it:

    Downloads|Resources
        Models
        Gallery

I will then configure the urls automatically to follow that structure as well (for SEO purposes). E.g. http://cress.soc.surrey.ac.uk/web/resources/models/housing-model


Please, let me know what do you think of this proposal and include your suggestions.


Regards,


David


P.S.: I am tracking this in the following issue: https://github.com/CRESS-Surrey/cress_website_code/issues/46

**Comment by drozas** [comment]
Feedback from Nigel:

This all looks good to me.

For an example of a model repository, see https://www.openabm.org/models

In addition to your suggested fields, we might add:

Required software (e.g. NetLogo, for NetLogo models, RePast for RePast models etc.)
Model version number and date

Nigel

**Comment by drozas** [comment]
Feedback from Jen:

​Can we also make it possible to have multiple files? For example, TELL ME would have 1 file (or link) for a zipped NetLogo + GIS data file, plus a documentation file (pdf).

**Comment by drozas** [comment]
Dear Nigel and Jen,

Thanks a lot for this. I think I have the requirements more clear now. I have, however, some questions to specify them a bit more:

>>> ​Can we also make it possible to have multiple files? For example, TELL ME would have 1 file (or link) for a zipped NetLogo + GIS data file, plus a documentation file (pdf).

Yes, that is not problem. Could you please provide me a list of the expected extensions for this? And the expected maximum file size?

>>> Required software (e.g. NetLogo, for NetLogo models, RePast for RePast models etc.)

Are you thinking on this just as another text field? Or rather as a selection list (a taxonomy)? If so, could you please specify the list of terms?


>>> Model version number and date

Regarding the model version, should a text field be sufficient for the model version?
Regarding the date, is this the published date (the one of the node, as in the example sent by Nigel), or an extra date field (e.g. the date in which the model was originally released?).

Thank you very much,

David

**Comment by drozas** [comment]
Textfield
>
>
> >>> Model version number and date
>
> Regarding the model version, should a text field be sufficient for the model version?

yes

> Regarding the date, is this the published date (the one of the node, as in the example sent by Nigel), or an extra date field (e.g. the date in which the model was originally released?).

Original date

(could have the published date too, but this could be automatically generated by the system, I suppose)

**Comment by drozas** [comment]
First version developed.

Pending issues: 
- Deploy changes in CSS by Nigel
- Clean up
- Images issue (permissions, to check by Nigel)

**Comment by drozas** [comment]
Dear Nigel and Jen,

I have just deployed a first version of this new functionality: http://cress.soc.surrey.ac.uk/web/resources/models
The models can be created by any staff member at http://cress.soc.surrey.ac.uk/web/node/add/model

I have left some "lorem ipsum" contents temporarily, so you can test it. Let me know once you are done, and I will delete them (or you can delete them as well if your prefer).

There are some issues pending:

    Nigel, could you please update the source code at dusty according to this diff (https://github.com/CRESS-Surrey/cress_website_code/commit/1fbffe0b1fd9b6ec148e9a370590c9535b419235). It is just a minor adjustment in the padding, again the menu layout was broken after adding the resources.
    The generated thumbnails of the system do not work (although the images are properly uploaded). I have been taking a look at this, and it seems it is a server permissions problem. Nigel, could you please check that the permissions configuration is as in https://www.drupal.org/node/244924 (Linux Servers section). If you need to test it, you will just need to create a test model with a picture. If the thumbnail is not created after saving the node, the permissions are still not properly set.
    Jen, I have added the following extensions for the files: zip, tar.gz, pdf, nlogo Let me know if there are any other necessary extensions (alternatively, I could install a contributed module to avoid this restriction, but that will require Nigel to deploy the source code and it might take a bit longer).

Kind Regards,

David

**Comment by drozas** [comment]
Images issues depends on #47 , I'll mark this as closed
